src/iac_scan_runner/api.py

```diff
@@ -167,15 +167,15 @@ async def post_scan(iac: UploadFile = File(..., description='IaC file (zip or ta
         return JSONResponse(status_code=status.HTTP_400_BAD_REQUEST, content=str(e))
         
 @app.get("/results", summary="Retrieve particular scan result by given uuid", responses={200: {}, 400: {"model": str}})
-async def get_scan_result(uuid: str) -> JSONResponse:
+async def get_scan_result(uuid: Optional[str]) -> JSONResponse:
     """
     Retrieve a particular scan result (GET method)
     :param uuid: Identifier of a scan record in database
     :return: JSONResponse object (with status code 200 or 400)
     """
     try:
         results_persistence = ResultsPersistence() 
-        if(uuid):           
+        if(uuid and uuid!=""):           
```

Comment:
```suggestion
        if uuid:           
```

Formatting is needed here, and also empty string is alreafy treated as False.

Reply:
Remove all redundant parentheses.

Reply:
Will be resolved later.